backend/lib/services/members/MemberManager.js

```diff
@@ -187,35 +187,38 @@ class MemberManager {
     this.subjectList.delete(item.id)
   }
 
-  async deleteServiceAccountSecret (item) {
+  async deleteServiceAccountSecrets (item) {
     const { namespace } = Member.parseUsername(item.id)
     if (namespace !== this.namespace) {
       throw new UnprocessableEntity('It is not possible to modify a ServiceAccount from another namespace')
     }
 
-    const name = _
+    const names = _
       .chain(item)
       .get('extensions.secrets', [])
-      .tap(secrets => {
-        if (secrets.length > 1) {
-          throw new UnprocessableEntity(`ServiceAccount ${namespace} has more than one secret`)
-        }
-      })
-      .head()
-      .get('name')
+      .map('name')
       .value()
-    await this.client.core.secrets.delete(namespace, name)
+
+    _.forEach(names, async name => {
+      await this.client.core.secrets.delete(namespace, name)
+    })
   }
 
   async getKubeconfig (item) {
     const { namespace, name } = Member.parseUsername(item.id)
-    const secretName = _
+    const secretNames = _
       .chain(item)
       .get('extensions.secrets')
+      .map('name')
+      .value()
+    const secrets = await this.client.core.secrets.list(namespace)
```

Comment:
At least we should use a field selector to reduce the resultSet of the list call: 
```sh
k get secrets --field-selector="type=kubernetes.io/service-account-token"
```